controllers/log/logr.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2021 Absa Group Limited
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+Generated by GoLic, for more details see: https://github.com/AbsaOSS/golic
+*/
+package log
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"github.com/rs/zerolog"
+)
+
+// Adapter implements logr.Logger interface.
+// The adapter allows us to encapsulate zerolog into go-logr/logr interface.
+type Adapter struct {
```

Comment:
Conceptual question: as we're implementing go-logr/logr interface here anyway (like https://github.com/go-logr/zapr does), would it make sense to continue using logr log semantics in the rest of the code instead of replacing it with zerolog-specific chained syntax (log.Info() -> logger.Info().Msg())

Reply:
@somaritane , good question, and appreciate review. 

First, controllerManager dosn't produce much logs. More or less it is during application start. The primary reason for adapter is to let ControllerManager produce logs same or similar style as the rest of the application. i.e. We don't want to controllerManager print simple message when `LOG_FORMAT=json`

 I'd like to use chained semantics, but zerolog and logr have not compatible syntax. 
logr has `WithValues(slice []interface{})` which adds some key-value pairs of context to a logger (it's slice where, odd keys, even values). Also `Info(msg string, keysAndValues ...interface{})`,  has key/value pairs which can then be used to add additional message information. So conversion is not 1:1. To Zerolog requires some additional logic to print those keyValues slices.

Technically, we can split into standalone `zerologr` package, but would be nice to do it when we will reuse package, do some init() magic, internalize etc.. I suggest to keep it as it is now and split when necessary.

Reply:
@kuritka , thank you for explanation.